app/controllers/nonsipp/memberreceivedpcls/WhatYouWillNeedPensionCommencementLumpSumController.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.memberreceivedpcls
+
+import controllers.actions.{AllowAccessActionProvider, DataRequiredAction, DataRetrievalAction, IdentifierAction}
+import models.NormalMode
+import models.SchemeId.Srn
+import navigation.Navigator
+import pages.nonsipp.memberpayments.WhatYouWillNeedPensionCommencementLumpSumPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewmodels.DisplayMessage.{ListMessage, ListType, Message, ParagraphMessage}
+import viewmodels.implicits._
+import viewmodels.models.{ContentPageViewModel, FormPageViewModel}
+import views.html.ContentPageView
+
+import javax.inject.{Inject, Named}
+
+class WhatYouWillNeedPensionCommencementLumpSumController @Inject()(
+  override val messagesApi: MessagesApi,
+  @Named("non-sipp") navigator: Navigator,
+  identify: IdentifierAction,
+  allowAccess: AllowAccessActionProvider,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
```

Comment:
Minor: these injections can be simplified with `identifyAndRequireData: IdentifyAndRequireData,` and then calling it like `identifyAndRequireData(srn) { implicit request =>`